### ISSUE TITLE
[BUG] UnitCastingInfo does not return true on Interruptible argument, when cast is interruptible 

### ISSUE THREAD

**Issue by Soothx** [JamminL/wotlk-classic-bugs#350]
#### 1.1. Description
When using `/dump UnitCastingInfo("target")` on a target that is currently casting, it returns information about the current cast. In 3.2, a boolean was added, indicating whether or not the spell being cast is interruptible, as seen in the source material. This boolean currently always returns `false`. 
In TBC, `/dump UnitCastingInfo("target")` has no 8th argument at all (Source 1), but on PTR the 8th argument is returned on dump, but will always return `false`, regardless of the type of target. I have tested casters in the open world, just targeting someone casting, using `/dump UnitCastingInfo("player")`, tested it in a duel, and it never returns anything but `false`.

  #### 1.2. How to Reproduce
           1. Target someone currently casting a spell that is interruptible.
           2. Type `/dump UnitCastingInfo("target")` in chat while having the target.
           3. See that the 8th argument always returns false, regardless of the cast.
  #### 1.3  Source Material
TBC (Source 1):
![image](https://user-images.githubusercontent.com/33608256/186617650-2cf5eb33-1d51-41f3-a23b-e91f08ab78b3.png)

PTR (Source 2):
![image](https://user-images.githubusercontent.com/33608256/186616640-ebd32fd8-c1d6-48f7-810f-f74a58631f87.png)

PTR `/dump UNIT_SPELLCAST_INTERRUPTIBLE` (Source 3):
![image](https://user-images.githubusercontent.com/33608256/186620975-f9397853-fd11-4a22-b1fe-2527f8be4d7c.png)

#### 2.1. Description
UnitCastingInfo was updated in patch 3.2 to include a boolean, containing information if a spellcast is interruptible or not. As Wrath Classic is running on 3.3.5, and the boolean is in the API, but always returns `false`, it should be fixed to return `true`.

Simply put, the expected behaviour when using `/dump UnitCastingInfo("target")` on a target that is currently casting an interruptible spell, is that it should return an 8th argument, that is `true`.

This can be used by various addons to show you whether or not a spellcast is interruptible, as it is in Retail.
  #### 2.2 Source Material
https://wowpedia.fandom.com/wiki/API_UnitCastingInfo
https://wowwiki-archive.fandom.com/wiki/API_UnitCastingInfo
https://wowpedia.fandom.com/wiki/UNIT_SPELLCAST_INTERRUPTIBLE

**Comment by raethkcj** [comment]
The return value is `notInterruptible`, not `interruptible`. So it's expected that it returns false for most player spells. You'd need to find a boss or an enemy with a physical cast to get it to return true.

**Comment by Soothx** [comment]
You are indeed correct, it returns true when a Paladins is casting during Divine Shield.
![image](https://user-images.githubusercontent.com/33608256/186658325-e0a0e551-90fd-4147-bb8b-3e4228c65870.png)
